core/src/data/interval/mod.rs

```diff
@@ -287,7 +287,7 @@ impl Interval {
                 format!("{:?}", to),
             )
             .into()),
-            (None, _) => Err(IntervalError::Unreachable.into()),
+            (_, _) => Err(IntervalError::UnsupportedExtractPart.into()),
```

Comment:
This is also not related to this issue.
We can remove this `UnsupportedExtractPart` error.